File: 509-fibonacci-number/509-fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        tab = [0, 1]
        if n <= 1:
            return tab[n]
        
        for i in range(2, n + 1):
            tab.append(tab[i - 1] + tab[i - 2])
            
        return tab[-1]
        

File: 509-fibonacci-number/509-fibonacci-number.java
class Solution {
    public int fib(int n) {
        if (n <= 1) return n;
    
        int[] tab = new int[n + 1];
        tab[0] = 0;
        tab[1] = 1;
        
        for (int i = 2; i <= n; ++i)
            tab[i] = tab[i - 1] + tab[i - 2];
        
        return tab[tab.length - 1];
    }
}